sdk/src/connections/target/rgb_sensor.cpp

```diff
@@ -580,3 +579,157 @@ aditof::Status RgbSensor::getInternalBuffer(uint8_t **buffer,
     return enqueueInternalBufferPrivate(buf);
 }
+
+void RgbSensor::setRw(int value) { this->Rw = value; }
+
+void RgbSensor::setGw(int value) { this->Gw = value; }
+
+void RgbSensor::setBw(int value) { this->Bw = value; }
+
+uint16_t RgbSensor::getValueFromData(uint8_t *pData, int x, int y, int width,
+                                     int height) {
+    return ((pData[x * width * 2 + y * 2 + 1] << 8) +
+            pData[x * width * 2 + y * 2]);
+}
+
+uint16_t RgbSensor::verticalKernel(uint8_t *pData, int x, int y, int width,
+                                   int height) {
+    int sum = 0;
+    int nr = 0;
+    if (x - 1 >= 0 && x - 1 < height) {
+        sum += getValueFromData(pData, x - 1, y, width, height);
+        nr++;
+    }
+    if (x + 1 >= 0 && x + 1 < height) {
+        sum += getValueFromData(pData, x + 1, y, width, height);
+        nr++;
+    }
+    return ((uint16_t)((float)sum / (float)nr));
+}
+uint16_t RgbSensor::horizontalKernel(uint8_t *pData, int x, int y, int width,
+                                     int height) {
+    int sum = 0;
+    int nr = 0;
+    if (y - 1 >= 0 && y - 1 < width) {
+        sum += getValueFromData(pData, x, y - 1, width, height);
+        nr++;
+    }
+    if (y + 1 >= 0 && y + 1 < width) {
+        sum += getValueFromData(pData, x, y + 1, width, height);
+        nr++;
+    }
+    return ((uint16_t)((float)sum / (float)nr));
+}
+uint16_t RgbSensor::plusKernel(uint8_t *pData, int x, int y, int width,
+                               int height) {
+    int sum = 0;
+    int nr = 0;
+    if (x - 1 >= 0 && x - 1 < height) {
+        sum += getValueFromData(pData, x - 1, y, width, height);
+        nr++;
+    }
+    if (x + 1 >= 0 && x + 1 < height) {
+        sum += getValueFromData(pData, x + 1, y, width, height);
+        nr++;
+    }
+    if (y - 1 >= 0 && y - 1 < width) {
+        sum += getValueFromData(pData, x, y - 1, width, height);
+        nr++;
+    }
+    if (y + 1 >= 0 && y + 1 < width) {
+        sum += getValueFromData(pData, x, y + 1, width, height);
+        nr++;
+    }
+    return ((uint16_t)((float)sum / (float)nr));
+}
+uint16_t RgbSensor::crossKernel(uint8_t *pData, int x, int y, int width,
+                                int height) {
+    int sum = 0;
+    int nr = 0;
+    if (x - 1 >= 0 && x - 1 < height && y - 1 >= 0 && y - 1 < width) {
+        sum += getValueFromData(pData, x - 1, y - 1, width, height);
+        nr++;
+    }
+    if (x + 1 >= 0 && x + 1 < height && y - 1 >= 0 && y - 1 < width) {
+        sum += getValueFromData(pData, x + 1, y - 1, width, height);
+        nr++;
+    }
+    if (x - 1 >= 0 && x - 1 < height && y + 1 >= 0 && y + 1 < width) {
+        sum += getValueFromData(pData, x - 1, y + 1, width, height);
+        nr++;
+    }
+    if (x + 1 >= 0 && x + 1 < height && y + 1 >= 0 && y + 1 < width) {
+        sum += getValueFromData(pData, x + 1, y + 1, width, height);
+        nr++;
+    }
+    return ((uint16_t)((float)sum / (float)nr));
```

Comment:
What if execution doesn't go through any of the IFs? Dividing 0 by 0 isn't desired.

Reply:
Also you could remove the round brackets at the edges. They don't do anything but make expression less readable.